### PR TITLE
Enable usage of streaming job worker via zbctl

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR enables usage of streaming via a new flag to the create worker command: `--streamEnabled`.

## Related issues

related #15552

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: clients/go/cmd/zbctl/main_test.go [hunk 1]
```diff
@@ -237,6 +248,39 @@ func (s *integrationTestSuite) AfterTest(_, _ string) {
...
 }
 
+// TestStreamingJobWorker is separate from TestCommonCommands because it requires us to first create the
+// stream, await until it exists, and then create jobs to push them out. Otherwise we couldn't really
+// assert we're testing streaming and not polling.
+func (s *integrationTestSuite) TestStreamingJobWorker() {
+	// given
+	workerName := uuid.NewString()
+	deployCommand := "--insecure deploy testdata/job_model.bpmn"
+	workerCommand := fmt.Sprintf("create --insecure worker jobType --streamEnabled --maxJobsHandle 1 --name %s --handler echo", workerName)
```
Comment (Zelldon) [inline clients/go/cmd/zbctl/main_test.go:258]: 🔧 Do you want to set polling to a higher value maybe?

File: clients/go/cmd/zbctl/main_test.go [hunk 2]
```diff
@@ -237,6 +248,39 @@ func (s *integrationTestSuite) AfterTest(_, _ string) {
...
+		if cmdOut, err := s.runCommand(strings.Fields(jobCommand), false); err != nil {
+			s.FailNowf("Failed while creating a job to push with command '%s' (%v). Output: \n%s", jobCommand, err, cmdOut)
+		}
+	}()
+
+	cmdOut, err := s.runCommand(strings.Fields(workerCommand), false)
+	s.NoErrorf(err, "Failed while executing worker command '%s'. Output: \n%s", workerCommand, cmdOut)
+
+	// then
+	s.assertGoldenFileMatchesOutput("testdata/create_worker.golden", false, "TestStreamingJobWorker", cmdOut)
```
Comment (Zelldon) [inline clients/go/cmd/zbctl/main_test.go:281]: 🤔 Where are these golden files? 🤔

**Review by ChrisKujawa — APPROVED** [review]
Great Thanks 👍🏼

File: clients/go/cmd/zbctl/main_test.go (hunk 2 above)
Comment (npepinpe) [reply clients/go/cmd/zbctl/main_test.go:281]: It's the same as those used by the table tests. They're in `cmd/zbctl/testdata`

File: clients/go/cmd/zbctl/main_test.go (hunk 1 above)
Comment (npepinpe) [reply clients/go/cmd/zbctl/main_test.go:258]: Oh, right, forgot :see_no_evil:

Comment (npepinpe) [reply clients/go/cmd/zbctl/main_test.go:258]: Oh right, forgot :smile:

**Comment by npepinpe** [comment]
bors merge

**Pull request merged**